modules/calc/src/main/java/com/opengamma/strata/calc/runner/function/result/CurrencyValuesArray.java

```diff
@@ -63,9 +64,42 @@
    * @return an instance with the specified currency and values
    */
   public static CurrencyValuesArray of(Currency currency, double[] values) {
+    return new CurrencyValuesArray(currency, values.clone());
+  }
+
+  /**
+   * Returns an instance with the specified currency and values without copying the values array.
+   * <p>
+   * This method reduces the overhead of creating an instance by not making a defensive copy of the values
+   * array at the expense of safety. The caller <strong>must not</strong> mutate the array of values after passing
+   * it to this constructor. Doing so would violate the immutability of this class.
+   *
+   * @param currency  the currency of the values
+   * @param values  the currency values
+   * @return an instance with the specified currency and values
+   */
+  public static CurrencyValuesArray ofUnsafe(Currency currency, double[] values) {
     return new CurrencyValuesArray(currency, values);
   }
 
+  /**
+   * Constructor that doesn't clone the input array to avoid unnecessary copies. The factory methods copy
+   * the input array where necessary.
+   *
+   * @param currency  the currency of the values
+   * @param values  the currency values
+   */
+  @ImmutableConstructor
+  private CurrencyValuesArray(
+      Currency currency,
+      double[] values) {
```

Comment:
A market data box with one value can be used with any number of scenarios. Think of that one value as the unperturbed value which is used in all scenarios. That's why there is a special case for 1.